hyperspy/io_plugins/hdf5.py

```diff
@@ -339,20 +339,40 @@ def dict2hdfgroup(dictionary, group, compression=None):
             group.attrs["_datetime_" + key] = repr(value)
         elif isinstance(value, list):
```

Comment:
It seems that a good fraction of the code for saving lists and tuples is the same, wouldn't it be possible to combine the two?